core/src/main/scala-2/sttp/tapir/internal/CodecValueClassMacro.scala

```diff
@@ -0,0 +1,37 @@
+package sttp.tapir.internal
+
+import sttp.tapir.Codec
+import sttp.tapir.CodecFormat.TextPlain
+
+import scala.reflect.macros.blackbox
+
+object CodecValueClassMacro {
+
+  def derivedValueClass[T: c.WeakTypeTag](c: blackbox.Context): c.Expr[Codec[String, T, TextPlain]] = {
+    import c.universe._
+
+    val tpe: Type = weakTypeOf[T]
+
+    val isValueClass: Boolean = {
+      import definitions._
+      tpe.typeSymbol.isClass && tpe <:< AnyValTpe && !ScalaPrimitiveValueClasses.contains(tpe.typeSymbol.asClass)
+    }
+
+    val constructor = tpe.decls.collectFirst { case m: MethodSymbol if m.isPrimaryConstructor => m }
+
+    val fields = constructor.get.paramLists.head
```

Comment:
I think that `fields` and `constructor` should be `lazy` to not process them in non value classes cases.

Reply:
i will put them in `else` branch then